### DISCUSSION TITLE
Cursors controller and Cursors View

### DISCUSSION THREAD

**Discussion by phamconganh** [opensheetmusicdisplay/opensheetmusicdisplay#979] [Ideas]
Why don't we create a CursorsController to control the iterator, some method like: next, nextTo(step), previous, .... and the interface for CursorView to can add multiple custom cursor view (like measureCursor, leftHandCursor, rightHandCursor)? The CursorsController has cursors = [] to handle the "update" method of the cursors view?

**Comment by rvilarl** [comment]
@phamconganh in regards to measureCursor have a look at the pull request #972 other types can be easily included

**Reply by phamconganh** [reply]
@rvilarl It is nice. And I try the measureCursor, leftHandCursor, rightHandCursor. But I want to create other cursors and inject them into OSMD, not just one.

**Reply by sschmidTU** [reply]
with the PR you can add multiple cursors. it's right in the title.

**Reply by phamconganh** [reply]
@rvilarl But each of them has one iterator. It would be more wonderful if all the view of cursors (measureCursor, leftHandCursor, rightHandCursor, verticalContainerStaffEntry) managed by one controller with one iterator. And if we want multiple iterators, we just clone the iterator from the controller and control itself. And we can inject lots of custom cursors view

**Reply by rvilarl** [reply]
@phamconganh sorry but I do not get the concept that you propose. I believe that it is good that each cursor has it iterator then from the application you can decide whether you move them at the same time or not. Could you also provide an image of the types of cursor that you refer to (measureCursor, left...) to make sure that we have the same understanding?

**Reply by sschmidTU** [reply]
I agree with rvilarl, if you want all cursors to always be at the same position (same iterator), you can just call next() on all cursors instead of only for one.
